Mono.Cecil.Cil/PortablePdb.cs

```diff
@@ -249,6 +249,12 @@ public void Dispose ()
 				return;
 
 			WritePdbHeap ();
+
+			if (module_metadata.string_heap != pdb_metadata.string_heap) {
```

Comment:
SetModuleMetadata is called for the embedded and external scenarios. PortablePdb.Dispose() returns early if embedded, so it should be unnecessary.